packages/next/src/server/app-render/app-render.tsx

```diff
@@ -329,7 +329,10 @@ async function generateFlight(
     const [MetadataTree, MetadataOutlet] = createMetadataComponents({
       tree: loaderTree,
       query,
-      metadataContext: createMetadataContext(urlPathname, ctx.renderOpts),
+      metadataContext: createMetadataContext(
+        url?.pathname ?? page,
```

Comment:
This seems incorrect? if page is sth ends with `/page` or `/route` but pathname is the one without it.